### PR TITLE
Refactor output context

### PR DESCRIPTION
**Pull request by camilbancioiu**
* Change `outputContext` to contain a `VMOutput` and write to it directly, simplifying the code
* Add function `outputContext.GetOutputAccount()`, which simplified code in `blockchainContext` and `storageContext`
* Overall adjustments to use the new `VMOutput` map fields (`OutputAccounts` and `StorageUpdates`)

### CODE REVIEW

File: arwen/contexts/output.go [hunk 1]
```diff
@@ -163,50 +156,39 @@ func (context *outputContext) FinishValue(value wasmer.Value) {
...
+	var currentLogEntry *vmcommon.LogEntry
+	currentLogEntry = nil
 
-	if _, ok := context.logs[strAdr]; !ok {
-		context.logs[strAdr] = logTopicsData{
-			topics: make([][]byte, 0),
-			data:   make([]byte, 0),
+	for _, logEntry := range context.outputState.Logs {
+		if bytes.Equal(logEntry.Address, address) {
+			currentLogEntry = logEntry
```
Comment (sasurobert) [inline arwen/contexts/output.go]: break

File: arwen/contexts/blockchain.go [hunk 2]
```diff
@@ -40,58 +40,57 @@ func (context *blockchainContext) NewAddress(creatorAddress []byte) ([]byte, err
...
+	if !isNew {
+		return outputAccount.Balance.Bytes()
 	}
 
-	balance, err := context.blockChainHook.GetBalance(addr)
+	balance, err := context.blockChainHook.GetBalance(address)
 	if err != nil {
 		fmt.Printf("GetBalance returned with error %s \n", err.Error())
+		if isNew {
+			context.host.Output().DeleteAccountFromOutput(address)
```
[the remark this answers is not in the document]
Comment (camilbancioiu) [reply arwen/contexts/blockchain.go]: Removed condition.

File: arwen/contexts/blockchain.go [hunk 3]
```diff
@@ -40,58 +40,57 @@ func (context *blockchainContext) NewAddress(creatorAddress []byte) ([]byte, err
...
+	if !isNew {
+		return outputAccount.Nonce, nil
 	}
 
-	nonce, err := context.blockChainHook.GetNonce(addr)
+	nonce, err := context.blockChainHook.GetNonce(address)
 	if err != nil {
 		fmt.Printf("GetNonce returned with error %s \n", err.Error())
+		if isNew {
+			context.host.Output().DeleteAccountFromOutput(address)
```
[the remark this answers is not in the document]
Comment (camilbancioiu) [reply arwen/contexts/blockchain.go]: Removed condition.

File: arwen/contexts/output.go (hunk 1 above)
Comment (camilbancioiu) [reply arwen/contexts/output.go]: Fixed.

**Pull request merged**